examples/chess_position.py

```diff
@@ -0,0 +1,159 @@
+
+occupancy_in_board = np.array([False, False, False, False, False, False,  True, False,
+                                True, False, False, False, False, False, False, False, 
+                               False,  True, False, False,  True,  True, False, False, 
+                               False,  True,  True, False,  True, False,  True,  True,
+                                True, False,  True,  True,  True, False, False, False,
+                               False, False, False,  True, False,  True, False,  True,
+                               False, False,  True, False,  True, False,  True, False,
+                                True, False, False, False,  True, False, False, False])
+
+# Variable to represent the whole board
+board = intvar(0, 12, (8,8), "board")
+
+# Variable to represent the pieces, with first row the probabilies and second row the indexes
+pieces = intvar(0, 10000000, (2, len(piece_probabilities)), name="pieces")
+
+# Make the probabilites into reasonable big values
+pieces_values = np.empty([len(piece_probabilities), len(piece_probabilities[0])], dtype=int)
+for i in range(len(piece_probabilities)):
+    for j in range(len(piece_probabilities[0])):
+        pieces_values[i,j] = int((piece_probabilities[i,j]*10000).item())
+
+# Assign values to pieces
+b, k, n, p, q, r = 0, 1, 2, 3, 4, 5 # Black pieces
+B, K, N, P, Q, R = 6, 7, 8, 9, 10, 11 # White pieces
+E = 12 # Empty square
+
+# Lowercases are black pieces, uppercases white pieces, and E is empty square
+dict = {0: 'b', 1: 'k', 2: 'n', 3: 'p', 4: 'q', 5: 'r', 6: 'B', 7: 'K', 8: 'N', 9: 'P', 10: 'Q', 11: 'R', 12: 'E'}
+
+# Find the minimum negative value along each row
+min_neg_row = np.apply_along_axis(lambda x: np.min(np.where(x < 0, x, np.inf)), 1, pieces_values)
+
+# Create a broadcastable array of minimum negative values
+min_neg_arr = np.repeat(min_neg_row[:, np.newaxis], pieces_values.shape[1], axis=1)
+
+# Shift all values in the array to be positive
+pieces_values = pieces_values - min_neg_arr
+
+pieces_values = pieces_values.astype(int)
+
+# All the constraints that a chess board has to follow
+model = Model(
+    # Both colors have only one king
+    (sum(board[l,c] == k for l in range(0,8) for c in range(0,8)) == 1),
+    (sum(board[l,c] == K for l in range(0,8) for c in range(0,8)) == 1),
+    # Both colors have maximum 16 pieces
+    (sum(board[l,c] < 6 for l in range(0,8) for c in range(0,8)) <= 16),
+    (sum(((board[l,c] >= 6) & (board[l,c] < 12)) for l in range(0,8) for c in range(0,8)) <= 16),
+    # Both colors have maximum 8 pawns
+    (sum(board[l,c] == p for l in range(0,8) for c in range(0,8)) <= 8),
+    (sum(board[l,c] == P for l in range(0,8) for c in range(0,8)) <= 8),
+    # Pawns can't be on the edge of the board
+    (sum(board[0,c] == p for c in range(0,8)) == 0),
+    (sum(board[0,c] == P for c in range(0,8)) == 0),
+    (sum(board[7,c] == p for c in range(0,8)) == 0),
+    (sum(board[7,c] == P for c in range(0,8)) == 0),
```

Comment:
In general the constraints can be written in a better way,  See previous comments